tests/pipelines/test_pipelines_automatic_speech_recognition.py

```diff
@@ -377,6 +379,33 @@ def test_chunking_fast(self):
         self.assertEqual(output, [{"text": ANY(str)}])
         self.assertEqual(output[0]["text"][:6], "ZBT ZC")
 
+    @require_torch
+    def test_chunking_fast_for_mctct(self):
+        model = MCTCTForCTC.from_pretrained("speechbrain/m-ctc-t-large")
+        processor = MCTCTProcessor.from_pretrained("speechbrain/m-ctc-t-large")
+        speech_recognizer = AutomaticSpeechRecognitionPipeline(
+            feature_extractor=processor.feature_extractor,
+            tokenizer=processor.tokenizer,
+            model=model,
+            chunk_length_s=10.0,
+            framework="pt",
+        )
+
+        ds = load_dataset("hf-internal-testing/librispeech_asr_dummy", "clean", split="validation").sort("id")
+        audio = ds[40]["audio"]["array"]
+
+        # It should work for a single non-batched input
+        output = speech_recognizer(audio)
+        self.assertEqual(output, {"text": ANY(str)})
+        self.assertEqual(output["text"][:26], "A man said to the universe")
+
+        # And also for multiple batched inputs
+        n_repeats = 2
+        audio_tiled = np.tile(audio, n_repeats)
+        output = speech_recognizer([audio_tiled], batch_size=2)
+        self.assertEqual(output, [{"text": ANY(str)}])
+        self.assertEqual(output[0]["text"][:26], "A man said to the universe")
```

Comment:
Flagging this because I can't for the life of me understand why the other ASR tests are constructed this way:

```python
n_repeats = 2
audio_tiled = np.tile(audio, n_repeats)
output = speech_recognizer([audio_tiled], batch_size=2)
```

If `audio` is an array with shape `(N,)`, then `audio_tiled` is an array with shape `(2 * N,)`, and `output` is a list of length 1. It seems to me like a correct test for batched inputs would instead look like:

```python
output = speech_recognizer([audio, audio], batch_size=2)
```

Which indeed works and produces an `output` which is length 2. Am I crazy? 😄

Reply:
Hmm - I'm not 100% sure here - was the idea to test a long input? cc'ing @Narsil here - was the test intended?

Reply:
This is for chunked batching.
The argument is `chunk_length_s=10.0,`

If you have a very long input audio, it **will** get chunked into parts. And so using `batch_size=n` should run the inference on the model on both chunks within 1 batch.

This test does test it.

Overall in pipeline, the size of the list doesn't matter for batching.